ICSharpCode.Decompiler/IL/ControlFlow/YieldReturnDecompiler.cs

```diff
@@ -1252,8 +1255,38 @@ private void CleanSkipFinallyBodies(ILFunction function)
 						entryPoint.Instructions.RemoveRange(1, entryPoint.Instructions.Count - 1);
 					}
 				}
+				entryPoint = AsyncAwaitDecompiler.GetBodyEntryPoint(container);
```

Comment:
Redundant assignment? `entryPoint` should already have this value?

Reply:
GetBodyEntryPoint returns the first block that does not have an unconditional branch as its first instruction. After the "skip finally" if-instruction is removed, we reuse the logic to get to the block that contains the generated finally-method.